projects/ngx-skeleton-loader/src/lib/ngx-skeleton-loader.component.ts

```diff
@@ -24,24 +27,41 @@ export class NgxSkeletonLoaderComponent implements OnInit, AfterViewInit, OnDest
   static ngAcceptInputType_animation: boolean | string;
 
   @Input()
-  count = 1;
+  count: NgxSkeletonLoaderConfig['count'];
 
   @Input()
-  loadingText = 'Loading...';
+  loadingText: NgxSkeletonLoaderConfig['loadingText'];
 
   @Input()
-  appearance: 'circle' | '' = '';
+  appearance: NgxSkeletonLoaderConfig['appearance'];
 
   @Input()
-  animation: 'progress' | 'progress-dark' | 'pulse' | 'false' | false = 'progress';
+  animation: NgxSkeletonLoaderConfig['animation'];
 
-  // This is required since ngStyle is using `any` as well
-  // More details in https://angular.io/api/common/NgStyle
-  // tslint:disable-next-line: no-any
-  @Input() theme: { [k: string]: any } = {};
+  @Input()
+  theme: NgxSkeletonLoaderConfigTheme;
 
   // tslint:disable-next-line: no-any
-  items: Array<any> = [];
+  items: Array<any>;
+
+  private config: NgxSkeletonLoaderConfig;
+
+  constructor(@Inject(NGX_SKELETON_LOADER_CONFIG) @Optional() config: NgxSkeletonLoaderConfig) {
```

Comment:
Doubt: shouldn't `config` be marked as optional?

```suggestion
  constructor(@Inject(NGX_SKELETON_LOADER_CONFIG) @Optional() config?: NgxSkeletonLoaderConfig) {
```

Reply:
I belive this is the goal of the `@Optional()` decorator. It actually does not matter to mark config as optional with the `?`. Note that I think the optional decorator was there before the optional property syntax was introduced to TypeScript.

In other words: you can use `@Operator()` to tell the DI framework that the parameter is optional. You can also use `?` marker but this is more a _TypesScript_ syntax (it does not provide any information to the Angular compiler which might actually fail IMO). This is pretty much the same as specifying config of type `NgxSkeletonLoaderConfig | undefined`.

Reply:
Well, the `@Optional()` as you said is used by the Angular compiler to make DI works properly, however the `?` is used to avoid a possibly mistake someone can do in TS, as for example access `config.count`. It will result in a compile-time error. And regarding to Angular compiler fail: no, it doesn'nt fail, it's even used a lot in the [**angular/components**](https://github.com/angular/components/blob/f2b94e8706f1683f91cca0e68bafe8caabc7681e/src/cdk-experimental/menu/menu-item-radio.ts#L45-L51):